### PR TITLE
Microservice PR

### PR DESCRIPTION
**Pull request by LuisARodr**
This PR has the following changes:
- Add the connection to the DB
- Add the repository to manage the Product
- Changes to convert the POJO into an entity

### CODE REVIEW

File: Back/src/main/java/com/nearsoft/BackApplication.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.web.support.SpringBootServletInitializer;
+
+@SpringBootApplication
+@EntityScan("com.nearsoft.domain.pojo")
+public class BackApplication extends SpringBootServletInitializer {
```
Comment (Joxebus) [inline Back/src/main/java/com/nearsoft/BackApplication.java]: This application doesn't have web part so you don't need to initialize a servlet.

File: Front/src/main/java/com/nearsoft/service/ProductService.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.nearsoft.messages;
+package com.nearsoft.service;
 
 import com.nearsoft.domain.pojo.Product;
```
Comment (Joxebus) [inline Front/src/main/java/com/nearsoft/service/ProductService.java]: Please rename the package of the `Product` to be just `com.nearsoft.domain`

File: Back/src/main/java/com/nearsoft/repository/ProductRepository.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public ProductRepository(EntityManager em) {
+        this.em = em;
+    }
+    public List<Product> selectProducts() {
+        StoredProcedureQuery spq = em.createStoredProcedureQuery("Select_Products", Product.class);
+
+        return spq.getResultList();
+    }
+
+    public void addProduct(String name, Double price, int stock) {
```
Comment (jass-holguin) [inline Back/src/main/java/com/nearsoft/repository/ProductRepository.java]: Is it better to pass each value separately or would it be better to pass the Product object?

File: pom.xml [hunk 4]
```diff
@@ -54,6 +54,15 @@
 			<artifactId>jstl</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
```
Comment (jass-holguin) [inline pom.xml]: do you really need this dependency? Are you using it ?

File: Commons/src/main/java/com/nearsoft/domain/pojo/Product.java [hunk 5]
```diff
@@ -36,4 +46,14 @@ public String getName() {
...
+    @Override
+    public String toString() {
+        return new StringJoiner(", ", Product.class.getSimpleName() + "[", "]")
+                .add("id=" + id)
+                .add("price=" + price)
+                .add("stock=" + stock)
+                .add("name='" + name + "'")
+                .toString();
+    }
 }
```
Comment (AndresSan6) [inline Commons/src/main/java/com/nearsoft/domain/pojo/Product.java]: Nitpick: Is considered a good practice to add a new line at the end of a file.
https://thoughtbot.com/blog/no-newline-at-end-of-file

File: pom.xml (hunk 4 above)
Comment (LuisARodr) [reply pom.xml]: Yes, ProductRepository is making use of it.

Comment (jass-holguin) [reply pom.xml]: yeah, I thought it would be enough with hibernate core and entity manager, but it seems those don't have the store procedure method :p !

**Pull request merged**